nexus/src/external_api/console_api.rs

```diff
@@ -135,6 +139,35 @@ pub async fn spoof_login_form(
     serve_console_index(rqctx.context()).await
 }
 
+/// Fetch the user associated with the current session
+#[endpoint {
+   method = GET,
+   path = "/session/me",
+}]
+pub async fn session_me(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+) -> Result<HttpResponseOk<views::User>, HttpError> {
```

Comment:
The `User` view currently just has identity metadata: name, id, timestamps, description (which probably doesn't make sense on there anyway). Eventually I assume this will have stuff like full name that we pull from the IdP.

Reply:
Today, the only actors are BuiltinUsers.  In the future this will be _at least_ ServiceAccounts and IdpUsers.  We might also keep BuiltinUsers but we might not.  The shapes of these things are potentially all different.  I'm a little leery of committing us further down the `BuiltinUsers` path (by exposing the view to every user) -- this endpoint will eventually need to return one of several possible structures.  (I don't feel strongly about this -- just worrying out loud.)